Rewrite the scan topic costmap plugins for multi-robot(namespace) before launch navigation. (#3572)
ros-planning/navigation2: nav2_bringup/README.md

* Some discussions about performance improvement of composed bringup could be found here: https://discourse.ros.org/t/nav2-composition/22175.

To use, please see the Nav2 [Getting Started Page](https://navigation.ros.org/getting_started/index.html) on our documentation website. Additional [tutorials will help you](https://navigation.ros.org/tutorials/index.html) go from an initial setup in simulation to testing on a hardware robot, using SLAM, and more.

Note:
* gazebo should be started with both libgazebo_ros_init.so and libgazebo_ros_factory.so to work correctly.
* spawn_entity node could not remap /tf and /tf_static to tf and tf_static in the launch file yet, used only for multi-robot situations. Instead it should be done as remapping argument <remapping>/tf:=tf</remapping>  <remapping>/tf_static:=tf_static</remapping> under ros2 tag in each plugin which publishs transforms in the SDF file. It is essential to differentiate the tf's of the different robot.

## Launch

### Multi-robot Simulation

> This is how to launch multi-robot simulation with simple command line. Please see the Nav2 documentation for further arugmehts.

Reviewer: ```suggestion
This is how to launch multi-robot simulation with simple command line. Please see the Nav2 documentation for further augments.
```

> #### cloned

Reviewer: ```suggestion
#### Cloned
```

> This allows to bring up the multiple robots cloning a single robot N times. And the parameter for single robot are loaded from `nav2_multirobot_params_all.yaml` file in default.

Reviewer: ```suggestion
This allows to bring up multiple robots, cloning a single robot N times at different positions in the map. The parameter are loaded from `nav2_multirobot_params_all.yaml` file by default.
```

> The multiple robots that consists of name and initial pose in YAML format will be launched based on command-line. The format for each robot is `robot_name={x: 0.0, y: 0.0, yaw: 0.0, roll: 0.0, pitch: 0.0, yaw: 0.0}`.

Reviewer: ```suggestion
The multiple robots that consists of name and initial pose in YAML format will be set on the command-line. The format for each robot is `robot_name={x: 0.0, y: 0.0, yaw: 0.0, roll: 0.0, pitch: 0.0, yaw: 0.0}`.
```
